Bali/src/main/java/OfferList.java

```diff
@@ -0,0 +1,117 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public class OfferList {
+    private List<Offer> offers;
+
+    public OfferList(){
+        offers = new ArrayList<>();
+    }
+
+    public Offer getOffer(String id){
+        for(Offer offer : offers){
+            if(offer.getId().equals(id)){
+                return offer;
+            }
+        }
+        System.out.println("Incorrect type of id!");
+        return null;
+    }
+
+    public static boolean validateOffer(Offer offer){
+        if(offer.getId() != null && offer.getId().length() != 0 &&
+        offer.getDescription() != null && offer.getDescription().length() != 0 &&
+        offer.getCreatedAt() != null && offer.getCreatedAt().length() != 0 &&
+        offer.getLink() != null && offer.getLink().length() != 0 &&
+        offer.getVendor() != null && offer.getVendor().length() != 0 &&
+        offer.getHashTags() != null && offer.getHashTags().size() != 0 && offer.getHashTags().size() <= 7 &&
+        offer.getValidUntil() != null && offer.getValidUntil().length() != 0 &&
+        offer.getPhotoLink() != null && offer.getPhotoLink().length() != 0 &&
+        offer.getDiscount() != null && offer.getDiscount().length() != 0 &&
+        offer.getReviews() != null && offer.getReviews().size() != 0){
+            return true;
+        }
+        return false;
+    }
+
+    public boolean addOffer(Offer offer){
+        if(OfferList.validateOffer(offer) && getOffer(offer.getId()) == null){
+            offers.add(offer);
+            return true;
+        }
+        return false;
+    }
+
+    public boolean editOffer(String id, Offer offer){
+        if(offer.getId() != null || offer.getVendor() != null || offer.getCreatedAt() != null){
```

Comment:
Indeed?) *Every* offer id is not null I suppose, then the method is never fulfilled